### PR TITLE
Bobax 16 modal Skeleton

### PR DESCRIPTION
**Pull request by SimonMTan**

### CODE REVIEW

File: src/pages/Home/index.tsx [hunk 1]
```diff
@@ -32,7 +41,18 @@ const Homepage = ({ order, setOrder }: OrderProps): JSX.Element => {
...
       <SearchBar searchQuery={searchQuery} setSearchQuery={setSearchQuery} />
-      <BobaContainer order={order} setOrder={setOrder} bobaList={filteredBobaList} />
+      <BobaContainer
+        order={order}
+        setOrder={setOrder}
+        bobaList={filteredBobaList}
+        setModalShow={setModalShow}
+        setBobaInfoModal={setBobaInfoModal}
+      />
+      <BobaCustomizeableModal
```
Comment (seintun) [inline src/pages/Home/index.tsx]: Use `<CustomizeBobaModal />`

File: src/types/common/main.d.ts [hunk 2]
```diff
@@ -23,3 +23,9 @@ export interface SearchBarProps {
   searchQuery: string;
   setSearchQuery: (searchQuery: string) => void;
 }
+
+export interface ModalProps {
```
Comment (seintun) [inline src/types/common/main.d.ts]: `CustomizeBobaModalProps`

File: src/components/Card/index.tsx [hunk 3]
```diff
@@ -5,24 +5,38 @@ import './index.css';
...
+  setBobaInfoModal,
+}: BobaCardProps): JSX.Element => {
   const { id, name, price, imageLink, description } = product;
 
   const handleAddItemToCart = () => {
     const cartList = [...order, product];
     setOrder(cartList);
   };
 
+  const customizeBoba = () => {
```
Comment (seintun) [inline src/components/Card/index.tsx]: `handleCustomizeBoba`

File: src/components/Card/index.tsx [hunk 4]
```diff
@@ -5,24 +5,38 @@ import './index.css';
 
 interface BobaCardProps extends OrderProps {
   product: BobaProps;
+  setModalShow: (modalShow: boolean) => void;
+  setBobaInfoModal: (bobaInfoModal: BobaProps) => void;
```
Comment (seintun) [inline src/components/Card/index.tsx]: `setCustomizationModal`

File: src/components/Card/index.tsx [hunk 5]
```diff
@@ -5,24 +5,38 @@ import './index.css';
...
   const { id, name, price, imageLink, description } = product;
 
   const handleAddItemToCart = () => {
     const cartList = [...order, product];
     setOrder(cartList);
   };
 
+  const customizeBoba = () => {
+    setModalShow(true);
+    setBobaInfoModal(product);
```
Comment (seintun) [inline src/components/Card/index.tsx]: `setCustomization`

File: src/components/Modal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+              <Form.Group>
+                <Form.Check
+                  type={'radio'}
+                  id={'default-radio'}
+                  label={'Organic Whole Milk'}
+                  value={'Organic Whole Milk'}
+                  onChange={(e) => {
+                    setMilk(e.target.value);
+                  }}
+                  checked={milk === 'Organic Whole Milk'}
```
Comment (seintun) [inline src/components/Modal/index.tsx]: Hope this is just temporary not hard coded.

File: src/components/Modal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+                />
+              </Form.Group>
+            </Col>
+            <Row>
+              <Col xs={12} md={8}>
+                Toppings(Optional)
+                <Form.Check
+                  type={'checkbox'}
+                  id={'default-checkbox'}
+                  label={'Boba($0.75)'}
```
Comment (seintun) [inline src/components/Modal/index.tsx]: Can we use existing topping objects from sampleBobaAPI?

File: src/components/Modal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+                  label={'Mango Jelly($0.75)'}
+                  value={'Mango Jelly'}
+                  onChange={(e) => {
+                    addTopping(e.target.value, e.target.checked);
+                  }}
+                  checked={toppings.some((t) => t.name === 'Mango Jelly')}
+                />
+                <Form.Check
+                  type={'checkbox'}
+                  id={'default-checkbox'}
```
Comment (seintun) [inline src/components/Modal/index.tsx]: Why id isn't unique? Should be utilized for which element is checked kinda like bobax-1, how it passed info for modal

**Review by seintun — CHANGES_REQUESTED** [review]

**Comment by seintun** [comment]
Build pipeline failing from eslint 
```3:54:00 PM: Failed to compile.
3:54:00 PM: 
3:54:00 PM: [eslint]
3:54:00 PM: src/components/Modal/index.tsx
3:54:00 PM:   Line 3:8:  'Col' is defined but never used  @typescript-eslint/no-unused-vars
3:54:00 PM:   Line 5:8:  'Row' is defined but never used  @typescript-eslint/no-unused-vars
3:54:00 PM: error Command failed with exit code 1. (https://ntl.fyi/exit-code-1)
3:54:00 PM: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
3:54:00 PM: ​
3:54:00 PM:   "build.command" failed                                        
3:54:00 PM: ────────────────────────────────────────────────────────────────
3:54:00 PM: ​
3:54:00 PM:   Error message
3:54:00 PM:   Command failed with exit code 1: yarn build (https://ntl.fyi/exit-code-1)
3:54:00 PM: ​
3:54:00 PM:   Error location
3:54:00 PM:   In Build command from Netlify app:
3:54:00 PM:   yarn build
3:54:00 PM: ​
```

File: src/components/Card/index.tsx (hunk 4 above)
Comment (SimonMTan) [reply src/components/Card/index.tsx]: this is for line 8 right?

Comment (seintun) [reply src/components/Card/index.tsx]: #9 since its talking about Modal

File: src/components/BobaContainer/index.tsx [hunk 9]
```diff
@@ -6,15 +6,30 @@ import './index.css';
 
 interface BobaContainerProps extends OrderProps {
   bobaList: BobaProps[];
+  setModalShow: (modalShow: boolean) => void;
+  setBobaInfoModal: (bobaInfoModal: BobaProps) => void;
```
Comment (seintun) [inline src/components/BobaContainer/index.tsx:10]: `setCustomizationModal`

**Review by seintun — APPROVED** [review]

**Pull request merged**